src/function/aggregate/holistic/quantile.cpp

```diff
@@ -93,9 +97,11 @@ struct QuantileOperation {
 		D_ASSERT(bind_data_p);
 		auto bind_data = (QuantileBindData *)bind_data_p;
 		auto v_t = (T *)state->v;
-		auto offset = (idx_t)((double)(state->pos - 1) * bind_data->quantile);
-		std::nth_element(v_t, v_t + offset, v_t + state->pos);
-		target[idx] = v_t[offset];
+		for (const auto &quantile : bind_data->quantiles) {
```

Comment:
This is always supposed to be a list of size 1, right? Perhaps assert that and explicitly only use the first entry?